### DISCUSSION TITLE
Where is the interpretation of an inheritance described for the business

### DISCUSSION THREAD

**Discussion by muricna** [eprocurementontology/eprocurementontology#666] [Q&A]
Is there a description as to how to interpret the arrow highlighted in yellow in the diagram below:

![image](https://github.com/user-attachments/assets/57a873fc-4a31-4c88-917b-7beb8e13d963)

**Comment by AchillesDougalis** [comment]
Predicate generalizations / specializations were added to the Conceptual model guide. 

![2024-07-23_14-01](https://github.com/user-attachments/assets/c1e163f0-855a-4680-8894-f7fa447f8f74)

**Comment by cristianvasquez** [comment]
related: https://github.com/OP-TED/ePO/discussions/632
related: https://github.com/OP-TED/ePO/issues/635

**Comment by cristianvasquez** [comment]
@AchillesDougalis thanks for the response. I'm concerned about how this is to be interpreted when producing SHACL shapes. We know that if a class inherits from another, all its properties are inherited. That's a convention. 

However, if class C1 has property P1, and class C2 which inherits from C1 has P2 which inherits from P1, what should the SHACL shape look like? should C2 have P2 and P1, or just P2?

**Comment by costezki** [comment]
The property inheritance is covered in the OWL core but not yet covered in the current [set of transformation rules for SHACL](https://meaningfy-ws.github.io/model2owl-docs/public-review/transformation/transf-rules2.html), nor it is included in the [Style Guide](https://semiceu.github.io/style-guide/1.0.0/gc-conceptual-model-conventions.html).

This is a new feature in the model2owl transformation script.

**Comment by AchillesDougalis** [comment]
@cristianvasquez , I believe that given the convention you mention, C2 should have P2 and P1.

**Comment by cristianvasquez** [comment]
@AchillesDougalis That's what I thought, 

But now I'm not sure if that was the intention in all the sub-properties in ePO

To shred some light and arrive a an informed convention, I believe a [list](https://github.com/OP-TED/ePO/issues/635) of examples of such properties will help

We can look at those concrete examples to check if always C2 with P2 and P1 makes sense, if it's such a convention is applicable or if we need something different than inheritance to connect such properties.

**Comment by cristianvasquez** [comment]
@costezki are there plans to add this in the [Style Guide](https://semiceu.github.io/style-guide/1.0.0/gc-conceptual-model-conventions.html) ?

**Reply by costezki** [reply]
I have just updated the description of https://github.com/OP-TED/model2owl/issues/205 which stipulates that StyleGuide, transformation rules and checkers must be written before any implementation takes place.
